pkgs/test_core/lib/src/runner/configuration/args.dart

```diff
@@ -174,6 +174,22 @@ String get usage => _parser.usage;
 /// Throws a [FormatException] if [args] are invalid.
 Configuration parse(List<String> args) => _Parser(args).parse();
 
+PathConfiguration _parsePathConfiguration(String option) {
+  final uri = Uri.parse(option);
+
+  final name = uri.queryParameters['name'];
+  final fullName = uri.queryParameters['full-name'];
+
+  return PathConfiguration(
+    filePath: uri.path,
+    testName: fullName != null
```

Comment:
And then pass the list of them here, which will mean we at least respect all options if passed, even though its probably going to be very rare to pass more than 1.